### PR TITLE
Dead types

### PR DESCRIPTION
**Pull request by samoht**
Kill all these types!

### CODE REVIEW

**Comment by samoht** [comment]
Side-effect of this commit: `mirage-types` does not need `io-page` anymore.

File: types/V1.mli [hunk 1]
```diff
@@ -16,194 +16,195 @@
...
-  (** Time operations for cooperative threads. *)
+  (** {1 Time operations for cooperative threads} *)
 
   type +'a io
-  (** A potentially blocking I/O operation *)
+  (** The type for potentially blocking I/O operation *)
 
   val sleep: float -> unit io
-  (** [sleep nsec] Block the current thread for [TODO remove float] *)
+  (** [sleep nsec] Block the current thread for. {b FIMXE:} remove float. *)
```
Comment (mor1) [inline types/V1.mli]: FIMXE -> FIXME :)

Comment (samoht) [inline types/V1.mli]: right, all add a `fixme` note to remember to fix it :p

File: types/V1.mli [hunk 2]
```diff
@@ -456,72 +438,72 @@ module type IP = sig
...
+  (** The type for callback functions that adds the UDP metadata for
+      [src] and [dst] IP addresses, the [src_port] of the connection
+      and the [buffer] payload of the datagram. *)
 
   val input: listeners:(dst_port:int -> callback option) -> t -> ipinput
-  (** [input listeners t] demultiplexes incoming datagrams based on their destination
-      port.  The [listeners] callback is will either return a concrete handler or
-      a [None], which results in the datagram being dropped. *)
+  (** [input listeners t] demultiplexes incoming datagrams based on
+      their destination port.  The [listeners] callback is will either
```
Comment (mor1) [inline types/V1.mli]: "is will" -> "will" ?

**Comment by samoht** [comment]
Actually we still need `DEVICE.error` as we still rely on the fact that implementation should have a proper `connect` function. That's annoying that we can't really enforce in the signature though.

**Pull request closed without merging**